src/e2ee/matrixKeyProvider.ts

```diff
@@ -44,7 +44,7 @@ export class MatrixKeyProvider extends BaseKeyProvider {
     encryptionKeyIndex: number,
     participantId: string,
   ): void => {
-    createKeyMaterialFromBuffer(encryptionKey).then(
+    createKeyMaterialFromBuffer(encryptionKey.buffer as ArrayBuffer).then(
```

Comment:
This warning is a bit annoying, a `.buffer` is not necessarly an ArrayBuffer? 
I wonder why we are having this warning only now... TBH I don't understand

At the end it is calling subtle crypto `importKey` that accepts An ArrayBuffer, a TypedArray, a DataView...

I don't know what is the best way to fix that.
We change our API to use ArrayBuffer? We call directly import key? or we request a change to `createKeyMaterialFromBuffer`

Reply:
So could be that:

```
   crypto.subtle.importKey('raw', encryptionKey, 'HKDF', false, [
      'deriveBits',
      'deriveKey',
    ]).then(
  ```
  
  If not what do we do if `encryptionKey.buffer` is not an ArrayBuffer? Maybe we could throw an error ?

Reply:
This seems like a good idea to me. I just added a commit patching this.